src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1947,8 +1947,59 @@ void c_typecheck_baset::typecheck_side_effect_function_call(
     if(symbol_table.symbols.find(identifier)==symbol_table.symbols.end())
     {
       // This is an undeclared function.
+
+      // Is it the polymorphic typed_target function ?
+      if(identifier == CPROVER_PREFIX "typed_target")
+      {
+        if(expr.arguments().size() != 1)
+        {
+          std::ostringstream error_message;
+          error_message << "expected 1 argument for "
+                        << CPROVER_PREFIX "typed_target"
+                        << " found " << expr.arguments().size();
+          throw invalid_source_file_exceptiont{
+            error_message.str(), expr.source_location()};
+        }
+
+        auto arg0 = expr.arguments().at(0);
```

Comment:
Nit pick: We don't commonly use `at`, and you've done the sanity check just before, so perhaps use `[0]` or `.front()` instead?

Reply:
fixed